tests/query_tests.rs

```diff
@@ -1,93 +1,319 @@
+            String::from(first_name)
+        })
+        .collect();
+
+    let expected_between_names = vec!["Dalia".to_string(), "Gilligan".to_string(), "Kevina".to_string(), "Meta".to_string(), "Noellyn".to_string()];
+
+    assert_eq!(names, expected_between_names);
+
+    // A query getting back elements having specific names
 
     let query_value = json!({
         "where": [
-            ["firstName", ">", "Kiara"],
-            ["firstName", "<", "Sam"]
+            ["firstName", "in", names]
         ],
         "startAt": 0,
         "limit": 100,
-        "orderBy": ["firstName", "asc"]
+        "orderBy": [
+            ["firstName", "asc"]
+        ]
     });
     let where_cbor = common::value_to_cbor(query_value, None);
-    let person_document_type = contract.document_types.get("person").expect("contract should have a person document type");
-    let query = DriveQuery::from_cbor(where_cbor.as_slice(), &contract, &person_document_type).expect("query should be built");
-    let (results, skipped) = query.execute_no_proof(&mut drive.grove, None).expect("proof should be executed");
-     assert_ne!(results.len(), 100);
-}
\ No newline at end of file
+    let person_document_type = contract
+        .document_types
+        .get("person")
+        .expect("contract should have a person document type");
+    let query = DriveQuery::from_cbor(where_cbor.as_slice(), &contract, &person_document_type)
+        .expect("query should be built");
+    let (results, skipped) = query
+        .execute_no_proof(&mut drive.grove, None)
+        .expect("proof should be executed");
+    let names: Vec<String> = results
+        .into_iter()
+        .map(|result| {
+            let document = Document::from_cbor(result.as_slice(), None, None).expect("we should be able to deserialize the cbor");
+            let first_name_value = document.properties.get("firstName").expect("we should be able to get the first name");
+            let first_name = first_name_value.as_text().expect("the first name should be a string");
+            String::from(first_name)
+        })
+        .collect();
+
+    assert_eq!(names, expected_between_names);
+
+    // A query getting back elements having specific names and over a certain age
+
+    let query_value = json!({
+        "where": [
+            ["firstName", "in", names],
+            ["age", ">=", 45]
+        ],
+        "startAt": 0,
+        "limit": 100,
+        "orderBy": [
+            ["firstName", "asc"],
+            ["age", "desc"]
+        ]
+    });
+    let where_cbor = common::value_to_cbor(query_value, None);
+    let person_document_type = contract
+        .document_types
+        .get("person")
+        .expect("contract should have a person document type");
+    let query = DriveQuery::from_cbor(where_cbor.as_slice(), &contract, &person_document_type)
+        .expect("query should be built");
+    let (results, skipped) = query
```

Comment:
drop unused variable
```suggestion
    let (results, _) = query
```